nav2_theta_star_planner/include/nav2_theta_star_planner/theta_star_planner.hpp

```diff
@@ -85,16 +64,19 @@ class ThetaStarPlanner : public nav2_core::GlobalPlanner
     const std::vector<coordsW> & raw_path,
     const double & dist_bw_points);
 
+protected:
+  std::shared_ptr<tf2_ros::Buffer> tf_;
+  rclcpp::Clock::SharedPtr clock_;
+  rclcpp::Logger logger_{rclcpp::get_logger("ThetaStarPlanner")};
+  std::string global_frame_, name_;
+
+  std::unique_ptr<theta_star::ThetaStar> planner_;
+
   /**
-   * @brief checks whether the start and goal points have costmap costs lower than LETHAL_COST
-   * @return the result of the check
+   * @brief the function responsible for calling the algorithm and retrieving a path from it
+   * @return global_path is the planned path to be taken
    */
-  bool isSafeToPlan() const
```

Comment:
Why remove this? That seemed like a very good test to make sure planning is even possible at all

Reply:
It has just been shifted the other class

Reply:
I see, can you point out where its called then from the planning request from the plugin? Just want to verify, from a second glance, I didn't see it

Reply:
https://github.com/Anshu-man567/navigation2/blob/e72f401219fc52d362e1c9dc5a2439b714a454d4/nav2_theta_star_planner/include/nav2_theta_star_planner/theta_star.hpp#L114
It is present on L114 of the theta_star.cpp file.

https://github.com/Anshu-man567/navigation2/blob/e72f401219fc52d362e1c9dc5a2439b714a454d4/nav2_theta_star_planner/src/theta_star_planner.cpp#L96
And is used on L96 of the theta_star_planner.cpp file

Reply:
isSafeToPlan returns true if in collision? Shouldn't that be the opposite?

Reply:
And if either aren't valid, why do linear interpolation? Shouldn't it return failure?

Reply:
Yeah it is the opposite, I do realise that. I wasn't able to think of a function name

Reply:
Well you could keep the name, just invert the return values on its use, eg returns `true` if valid and `false` if not valid.

Reply:
I have changed its name to `isUnsafeToPlan()`

Reply:
That works too